tst/com/amazon/corretto/crypto/provider/test/AesTest.java

```diff
@@ -1142,61 +1188,110 @@ public void safeReuse() throws Throwable {
     assertFalse((boolean) sneakyGetField(spi, "contextInitialized"));
     assertNull(sneakyGetField(spi, "context"));
     byte[] ciphertext1 = c.doFinal(plaintext1);
+    if (saveNativeContext(spi)) {
+      assertNotNull(sneakyGetField(spi, "context"));
+    } else {
+      assertNull(sneakyGetField(spi, "context"));
+    }
     c.init(Cipher.ENCRYPT_MODE, key, spec2);
     assertFalse((boolean) sneakyGetField(spi, "contextInitialized"));
-    assertNull(sneakyGetField(spi, "context"));
     byte[] ciphertext2 = c.doFinal(plaintext2);
+    if (saveNativeContext(spi)) {
+      assertNotNull(sneakyGetField(spi, "context"));
+    } else {
+      assertNull(sneakyGetField(spi, "context"));
+    }
     c.init(Cipher.ENCRYPT_MODE, key, spec3);
     assertFalse((boolean) sneakyGetField(spi, "contextInitialized"));
-    assertNotNull(sneakyGetField(spi, "context"));
     byte[] ciphertext3 = c.doFinal(plaintext3);
-
     c.init(Cipher.DECRYPT_MODE, key, spec1);
     assertFalse((boolean) sneakyGetField(spi, "contextInitialized"));
-    assertNotNull(sneakyGetField(spi, "context"));
     assertArrayEquals(plaintext1, c.doFinal(ciphertext1));
+    if (saveNativeContext(spi)) {
+      assertNotNull(sneakyGetField(spi, "context"));
+    } else {
+      assertNull(sneakyGetField(spi, "context"));
+    }
+
     c.init(Cipher.DECRYPT_MODE, key, spec2);
     assertFalse((boolean) sneakyGetField(spi, "contextInitialized"));
-    assertNotNull(sneakyGetField(spi, "context"));
     assertArrayEquals(plaintext2, c.doFinal(ciphertext2));
+    if (saveNativeContext(spi)) {
+      assertNotNull(sneakyGetField(spi, "context"));
+    } else {
+      assertNull(sneakyGetField(spi, "context"));
+    }
+
```

Comment:
very minor nit: would it make sense to pull this (heavily duplicated) assertion logic into a `assertNativeContextOk` function or something?

Reply:
Thanks for the suggestion.